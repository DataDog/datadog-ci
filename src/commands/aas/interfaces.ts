/**
 * Configuration options provided by the user through
 * the CLI in order to instrument properly.
 */
export interface AasConfigOptions {
  // AAS Targeting options
  subscriptionId?: string
  resourceGroup?: string
  aasName?: string
  resourceIds?: string[]

  // Configuration options
  service?: string
  environment?: string
  version?: string
  isInstanceLoggingEnabled?: boolean
  isProfilingEnabled?: boolean
  logPath?: string
  isDotnet?: boolean
  // no-dd-sa:typescript-best-practices/boolean-prop-naming
  shouldNotRestart?: boolean
}
