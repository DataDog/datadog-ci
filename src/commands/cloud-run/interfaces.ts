/**
 * Summarize relevant information about a Cloud Run log.
 * Also used to build CSV log files.
 * @typedef {Object} CloudRunLog
 * @property {string} severity - The level of severity of the log. It can be values such as 'DEBUG', 'INFO', 'ERROR', etc.
 * @property {string} timestamp - The timestamp of when the log was generated.
 * @property {string} logName - The name of the log
 * @property {string} message - The actual log message detailing what event occurred.
 */
export interface CloudRunLog {
  severity: string
  timestamp: string
  logName: string
  message: string
}

/**
 * Contains all the information used to create a log file.
 * @typedef {Object} LogConfig
 * @property {string} type - string name of the type of log. Used when printing CLI messages.
 * @property {string} fileName - The name of the log file (such as 'all_logs.csv' or 'error_logs.csv')
 * @property {string} [severityFilter] - Optional filter to modify the Logging query. Example: ' AND severity="DEBUG"'
 */
export interface LogConfig {
  type: string
  fileName: string
  severityFilter?: string
}
