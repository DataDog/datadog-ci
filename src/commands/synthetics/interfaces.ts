import {Metadata} from '../../helpers/interfaces'
import {ProxyConfiguration} from '../../helpers/utils'

interface Timings {
  dns: number
  download: number
  firstByte: number
  ssl: number
  tcp: number
  total: number
}

export interface Result {
  device: {
    id: string
  }
  duration?: number
  error?: string
  errorCode?: string
  errorMessage?: string
  eventType: string
  passed: boolean
  stepDetails: Step[]
  timings?: Timings
  unhealthy?: boolean
}

export interface PollResult {
  dc_id: number
  result: Result
  resultID: string
}

interface Resource {
  duration: number
  size: number
  type: string
  url: string
}

export interface Step {
  apmTraceIds: string[]
  browserErrors: string[]
  description: string
  duration: number
  error?: string
  resource: Resource
  screenshotBucketKey: boolean
  skipped: boolean
  snapshotBucketKey: boolean
  stepId: number
  type: string
  url: string
  value: string
}

export interface Test {
  config: {
    assertions: Assertion[]
    request: {
      headers: {[key: string]: string}
      method: string
      timeout: number
      url: string
    }
    variables: string[]
  }
  created_at: string
  created_by: User
  locations: string[]
  message: string
  modified_at: string
  modified_by: User
  monitor_id: number
  name: string
  options: {
    ci?: {
      executionRule: ExecutionRule
    }
    device_ids: string[]
    min_failure_duration: number
    min_location_failed: number
    tick_every: number
  }
  overall_state: number
  overall_state_modified: string
  public_id: string
  status: string
  stepCount: number
  tags: string[]
  type: string
}

export interface Assertion {
  actual: string | number | Date | {[key: string]: any}
  errorMessage?: string
  operator: Operator
  property?: string
  target: string | number | Date | {[key: string]: any}
  type: string
  valid: boolean
}

export enum Operator {
  contains = 'contains',
  doesNotContain = 'doesNotContain',
  is = 'is',
  isNot = 'isNot',
  lessThan = 'lessThan',
  matches = 'matches',
  doesNotMatch = 'doesNotMatch',
  validates = 'validates',
  isInMoreThan = 'isInMoreThan',
  isInLessThan = 'isInLessThan',
}

interface User {
  email: string
  handle: string
  id: number
  name: string
}

export interface TriggerResponse {
  device: string
  location: number
  public_id: string
  result_id: string
}

export interface TriggerResult extends TriggerResponse {
  pollingTimeout: number
  result?: PollResult
}

interface Location {
  display_name: string
  id: number
  is_active: boolean
  name: string
  region: string
}

export interface LocationsMapping {
  [key: number]: string
}

export interface Trigger {
  locations: Location[]
  results: TriggerResponse[]
  triggered_check_ids: string[]
}

interface RetryConfig {
  count: number
  interval: number
}

export interface ConfigOverride {
  allowInsecureCertificates?: boolean
  basicAuth?: BasicAuthCredentials
  body?: string
  bodyType?: string
  cookies?: string
  deviceIds?: string[]
  executionRule?: ExecutionRule
  followRedirects?: boolean
  headers?: {[key: string]: string}
  locations?: string[]
  pollingTimeout?: number
  retry?: RetryConfig
  startUrl?: string
  variables?: {[key: string]: string}
}

export interface Payload extends ConfigOverride {
  metadata?: Metadata
  public_id: string
}

interface BasicAuthCredentials {
  password: string
  username: string
}

export interface TemplateContext extends NodeJS.ProcessEnv {
  DOMAIN: string
  HOST: string
  HOSTNAME: string
  ORIGIN: string
  PARAMS: string
  PATHNAME: string
  PORT: string
  PROTOCOL: string
  SUBDOMAIN: string | undefined
  URL: string
}

export interface TriggerConfig {
  config: ConfigOverride
  id: string
}

export enum ExecutionRule {
  BLOCKING = 'blocking',
  NON_BLOCKING = 'non_blocking',
  SKIPPED = 'skipped',
}

export interface Suite {
  tests: TriggerConfig[]
}

export interface TestSearchResult {
  tests: {
    public_id: string
  }[]
}

export interface APIHelper {
  getTest(testId: string): Promise<Test>
  pollResults(resultIds: string[]): Promise<{results: PollResult[]}>
  searchTests(query: string): Promise<TestSearchResult>
  triggerTests(testsToTrigger: Payload[]): Promise<Trigger>
}

export interface APIConfiguration {
  apiKey: string
  appKey: string
  baseIntakeUrl: string
  baseUrl: string
  proxyOpts: ProxyConfiguration
}
