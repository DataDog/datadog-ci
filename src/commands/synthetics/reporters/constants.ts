import chalk from 'chalk'

export const ICONS = {
  FAILED: chalk.bold.red('✖'),
  FAILED_NON_BLOCKING: chalk.bold.yellow('✖'),
  SKIPPED: chalk.bold.yellow('⇢'),
  SUCCESS: chalk.bold.green('✓'),
}
