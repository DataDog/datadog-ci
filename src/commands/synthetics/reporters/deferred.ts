import {Writable} from 'stream'

import {DefaultReporter} from './default'

export class DeferredReporter extends DefaultReporter {
  private buffer = ''

  constructor() {
    super({
      context: {
        stderr: process.stderr,
        stdout: new Writable({
          write: (chunk: Buffer, encoding, callback) => {
            this.buffer += chunk.toString()
            callback()
          },
        }),
      },
    })
  }

  public flush(): void {
    process.stdout.write('\n')
    process.stdout.write(this.buffer)
  }
}
