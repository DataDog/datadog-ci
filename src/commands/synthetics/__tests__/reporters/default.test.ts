jest.unmock('chalk')

import {BaseContext} from 'clipanion/lib/advanced'

import {ExecutionRule, MainReporter, Result, Summary, Test, UserConfigOverride} from '../../interfaces'
import {DefaultReporter} from '../../reporters/default'
import {DEFAULT_COMMAND_CONFIG} from '../../run-tests-command'

import {
  getApiResult,
  getApiTest,
  getFailedBrowserResult,
  getSummary,
  getTimedOutBrowserResult,
  MOCK_BASE_URL,
} from '../fixtures'

/**
 * A good amount of these tests rely on Jest snapshot assertions.
 * If you make some changes in the output of the default reporter, chances are you
 * will also have to update the snapshots from `./__snapshots__/default.test.ts.snap`.
 * To do that, you can run the following command: `yarn test --updateSnapshot reporters/default.test.ts`.
 * More information on https://jestjs.io/docs/snapshot-testing.
 */

describe('Default reporter', () => {
  const baseUrlFixture = 'https://app.datadoghq.com/'
  const writeMock = jest.fn()
  const mockContext: unknown = {
    context: {
      stdout: {
        write: writeMock,
      },
    },
  }
  const reporter = new DefaultReporter(mockContext as {context: BaseContext})

  it('should log for each hook', () => {
    type ReporterCall = {[Fn in keyof MainReporter]: [Fn, Parameters<MainReporter[Fn]>, number]}[keyof MainReporter]

    // `testWait`/`resultReceived` is skipped as nothing is logged for the default reporter.
    const calls: ReporterCall[] = [
      ['error', ['error'], 1],
      ['initErrors', [['error']], 1],
      ['log', ['log'], 1],
      ['reportStart', [{startTime: 0}], 1],
      ['resultEnd', [getApiResult('1', getApiTest()), ''], 1],
      ['runEnd', [getSummary(), ''], 1],
      ['testTrigger', [getApiTest(), '', ExecutionRule.BLOCKING, {}], 1],
      ['testsWait', [[getApiTest()], '', ''], 2],
    ]

    for (const [fnName, args, calledTimes] of calls) {
      ;(reporter[fnName] as any)(...args)
      expect(writeMock).toHaveBeenCalledTimes(calledTimes)
      writeMock.mockClear()
    }
  })

  describe('testTrigger', () => {
    beforeEach(() => {
      writeMock.mockClear()
    })

    const testObject: Pick<Test, 'name'> = {
      name: 'Request on example.org',
    }
    const testId = 'aaa-bbb-ccc'

    const cases: [string, ExecutionRule, UserConfigOverride][] = [
      ['Blocking test, without config overwrite', ExecutionRule.BLOCKING, {}],
      ['Blocking test, with 1 config override', ExecutionRule.BLOCKING, {startUrl: 'foo'}],
      ['Blocking test, with 2 config overrides', ExecutionRule.BLOCKING, {startUrl: 'foo', body: 'hello'}],
      ['Non-blocking test from Datadog, without config overwrite', ExecutionRule.NON_BLOCKING, {}],
      ['Non-blocking test from Datadog, with 1 config override', ExecutionRule.NON_BLOCKING, {startUrl: 'foo'}],
      [
        'Non-blocking test from Datadog, with 2 config overrides',
        ExecutionRule.NON_BLOCKING,
        {startUrl: 'foo', body: 'hello'},
      ],
      [
        'Non-blocking test, with 1 config override',
        ExecutionRule.NON_BLOCKING,
        {executionRule: ExecutionRule.NON_BLOCKING},
      ],
      [
        'Non-blocking test, with 2 config overrides',
        ExecutionRule.NON_BLOCKING,
        {startUrl: 'foo', executionRule: ExecutionRule.NON_BLOCKING},
      ],
      ['Skipped test, with 1 config override', ExecutionRule.SKIPPED, {executionRule: ExecutionRule.SKIPPED}],
      [
        'Skipped test, with 2 config overrides',
        ExecutionRule.SKIPPED,
        {startUrl: 'foo', executionRule: ExecutionRule.SKIPPED},
      ],
      ['Skipped test from Datadog, without config overwrite', ExecutionRule.SKIPPED, {}],
      ['Skipped test from Datadog, with 1 config override', ExecutionRule.SKIPPED, {startUrl: 'foo'}],
      ['Skipped test from Datadog, with 2 config overrides', ExecutionRule.SKIPPED, {startUrl: 'foo', body: 'hello'}],
    ]

    test.each(cases)('%s', (title, executionRule, config) => {
      reporter.testTrigger(testObject, testId, executionRule, config)
      const mostRecentOutput = writeMock.mock.calls[writeMock.mock.calls.length - 1][0]
      expect(mostRecentOutput).toMatchSnapshot()
    })
  })

  test('testsWait outputs triggered tests', async () => {
    reporter.testsWait(new Array(11).fill(getApiTest()), baseUrlFixture, '123')
    const output = writeMock.mock.calls.map((c) => c[0]).join('\n')
    expect(output).toMatchSnapshot()
  })

  describe('resultEnd', () => {
    const createApiResult = (
      resultId: string,
      passed: boolean,
      executionRule = ExecutionRule.BLOCKING,
      test: Test
    ): Result => {
      const errorMessage = JSON.stringify([
        {
          actual: 1234,
          operator: 'lessThan',
          target: 1000,
          type: 'responseTime',
        },
      ])
      const failure = {code: 'INCORRECT_ASSERTION', message: errorMessage}

      const result = getApiResult(resultId, test)

      result.executionRule = executionRule
      result.passed = passed
      result.result = {...result.result, ...(passed ? {} : {failure}), passed}

      return result
    }

    const apiTest = getApiTest('aaa-aaa-aaa')
    const cases = [
      {
        description: '1 API test, 1 location, 1 result: success',
        fixtures: {
          baseUrl: baseUrlFixture,
          results: [getApiResult('1', apiTest)],
        },
      },
      {
        description: '1 API test, 1 location, 3 results: success, failed non-blocking, failed blocking',
        fixtures: {
          baseUrl: baseUrlFixture,
          results: [
            createApiResult('1', true, ExecutionRule.BLOCKING, apiTest),
            createApiResult('2', false, ExecutionRule.NON_BLOCKING, apiTest),
            createApiResult('3', false, ExecutionRule.BLOCKING, apiTest),
          ],
        },
      },
      {
        description: '3 Browser test: failed blocking, timed out, global failure',
        fixtures: {
          baseUrl: baseUrlFixture,
          results: [
            getFailedBrowserResult(),
            getTimedOutBrowserResult(),
            {
              ...getTimedOutBrowserResult(),
              result: {
                duration: 0,
                failure: {code: 'FAILURE_CODE', message: 'Failure message'},
                passed: false,
                steps: [],
              },
              timedOut: false,
            },
          ],
        },
      },
    ]

    test.each(cases)('$description', (testCase) => {
      const {results, baseUrl} = testCase.fixtures
      for (const result of results) {
        reporter.resultEnd(result, baseUrl)
      }
      const output = writeMock.mock.calls.map((c) => c[0]).join('')
      expect(output).toMatchSnapshot()
    })
  })

  describe('runEnd', () => {
    beforeEach(() => {
      writeMock.mockClear()
      jest.useFakeTimers()
      reporter.reportStart({startTime: Date.now() - 567890}) // 9m 28s
    })

    const baseSummary: Summary = getSummary()

    const complexSummary: Summary = {
      batchId: 'batch-id',
      criticalErrors: 2,
      failed: 1,
      failedNonBlocking: 3,
      passed: 2,
      skipped: 1,
      testsNotFound: new Set(['ccc-ccc-ccc', 'ddd-ddd-ddd']),
      timedOut: 1,
    }

    const cases: {description: string; summary: Summary}[] = [
      {
        description: 'Simple case with 1 test with 1 result (passed)',
        summary: {...baseSummary, passed: 1},
      },
      {
        description: 'Complex case with all the tests and results outcomes possible',
        summary: complexSummary,
      },
      {
        description: 'Case where some outcomes are empty or missing',
        summary: {
          ...baseSummary,
          criticalErrors: 1,
          failedNonBlocking: 1,
          passed: 3,
          testsNotFound: new Set(['bbb-bbb-bbb']),
        },
      },
    ]

    test.each(cases)('$description', (testCase) => {
      reporter.runEnd(testCase.summary, MOCK_BASE_URL)
      const mostRecentOutput = writeMock.mock.calls[writeMock.mock.calls.length - 1][0]
      expect(mostRecentOutput).toMatchSnapshot()
    })

    const onDemandConcurrencyCaps: {description: string; cap: number}[] = [
      {cap: 0, description: 'does not communicate for uncapped orgs'},
      {cap: 1, description: 'communicates no parallelization (1 test at a time)'},
      {cap: 2, description: 'communicates 2 tests parallelization'},
    ]

    test.each(onDemandConcurrencyCaps)('$description', (testCase) => {
      reporter.runEnd({...baseSummary, passed: 1}, MOCK_BASE_URL, {onDemandConcurrencyCap: testCase.cap})
      const mostRecentOutput = writeMock.mock.calls[writeMock.mock.calls.length - 1][0]
      expect(mostRecentOutput).toMatchSnapshot()
    })
  })
})
