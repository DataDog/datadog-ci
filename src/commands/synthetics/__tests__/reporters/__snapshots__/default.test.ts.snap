// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default reporter resultEnd 1 API test, 1 location, 1 result: success 1`] = `
"[1m[32mâœ“[39m[22m [[1m[2maaa-aaa-aaa[22m[22m] [1mTest name[22m - [1m[32mlocation: [1mFrankfurt (AWS)[22m[1m[39m[22m
  âŽ‹ Total duration: 123 ms - View test run details: [2m[36mhttps://app.datadoghq.com/synthetics/details/aaa-aaa-aaa?resultId=1&from_ci=true[39m[22m 
  [1m[32mâœ“[39m[22m [1m[32m[1mGET[22m[1m - http://fake.url[39m[22m

"
`;

exports[`Default reporter resultEnd 1 API test, 1 location, 3 results: success, failed non-blocking, failed blocking 1`] = `
"[1m[32mâœ“[39m[22m [[1m[2maaa-aaa-aaa[22m[22m] [1mTest name[22m - [1m[32mlocation: [1mFrankfurt (AWS)[22m[1m[39m[22m
  âŽ‹ Total duration: 123 ms - View test run details: [2m[36mhttps://app.datadoghq.com/synthetics/details/aaa-aaa-aaa?resultId=1&from_ci=true[39m[22m 
  [1m[32mâœ“[39m[22m [1m[32m[1mGET[22m[1m - http://fake.url[39m[22m

[1m[33mâœ–[39m[22m [[1m[33mnon-blocking[39m[22m] [[1m[2maaa-aaa-aaa[22m[22m] [1mTest name[22m - [1m[33mlocation: [1mFrankfurt (AWS)[22m[1m[39m[22m
  âŽ‹ Total duration: 123 ms - View test run details: [2m[36mhttps://app.datadoghq.com/synthetics/details/aaa-aaa-aaa?resultId=2&from_ci=true[39m[22m 
  [1m[33mâœ–[39m[22m [1m[33m[1mGET[22m[1m - http://fake.url[39m[22m
[1m[33m  - Assertion failed:[39m[22m
[1m[33m    â–¶ responseTime should be less than [4m1000[24m. Actual: [4m1234[24m[39m[22m

[1m[31mâœ–[39m[22m [[1m[31mblocking[39m[22m] [[1m[2maaa-aaa-aaa[22m[22m] [1mTest name[22m - [1m[31mlocation: [1mFrankfurt (AWS)[22m[1m[39m[22m
  âŽ‹ Total duration: 123 ms - View test run details: [2m[36mhttps://app.datadoghq.com/synthetics/details/aaa-aaa-aaa?resultId=3&from_ci=true[39m[22m 
  [1m[31mâœ–[39m[22m [1m[31m[1mGET[22m[1m - http://fake.url[39m[22m
[1m[31m  - Assertion failed:[39m[22m
[1m[31m    â–¶ responseTime should be less than [4m1000[24m. Actual: [4m1234[24m[39m[22m

"
`;

exports[`Default reporter resultEnd 3 Browser test: failed blocking, timed out, global failure 1`] = `
"[1m[31mâœ–[39m[22m [[1m[31mblocking[39m[22m] [[1m[2mabc-def-ghi[22m[22m] [1mTest name[22m - [1m[31mlocation: [1mLocation name[22m[1m[39m[22m - [1m[31mdevice: [1mchrome.laptop_large[22m[1m[39m[22m
  âŽ‹ Total duration: 20000 ms - View test run details: [2m[36mhttps://app.datadoghq.com/synthetics/details/abc-def-ghi/result/1?from_ci=true[39m[22m 
    [1m[32mâœ“[39m[22m | [1m1000[22mms - Navigate to start URL
    [2mhttps://example.org/[22m
    [1m[31mâœ–[39m[22m | [1m1000[22mms - Navigate
    [2mhttps://example.org/[22m
    [31m[2mNavigation failure[22m[39m
    [1m[31mâœ–[39m[22m | [1m1000[22mms - Assert
    [2mvalue[22m
    [31m[2mStep failure[22m[39m
    [1m[33mâ‡¢[39m[22m | 3 skipped steps

[1m[31mâœ–[39m[22m [[1m[31mblocking[39m[22m] [[1m[2mabc-def-ghi[22m[22m] [1mTest name[22m - [1m[31mlocation: [1mLocation name[22m[1m[39m[22m
  âŽ‹ View test run details: [2m[36mhttps://app.datadoghq.com/synthetics/details/abc-def-ghi/result/1?from_ci=true[39m[22m (not yet received)
[31m    [[1mTIMEOUT[22m] - [2mResult timed out[22m[39m

[1m[31mâœ–[39m[22m [[1m[31mblocking[39m[22m] [[1m[2mabc-def-ghi[22m[22m] [1mTest name[22m - [1m[31mlocation: [1mLocation name[22m[1m[39m[22m
  âŽ‹ View test run details: [2m[36mhttps://app.datadoghq.com/synthetics/details/abc-def-ghi/result/1?from_ci=true[39m[22m 
[31m    [[1mFAILURE_CODE[22m] - [2mFailure message[22m[39m

"
`;

exports[`Default reporter runEnd Case where some outcomes are empty or missing 1`] = `
"[33m[1m1[22m test not found[39m [90m(bbb-bbb-bbb)[39m
View full summary in Datadog: [2m[36mhttps://app.datadoghq.com/synthetics/explorer/ci?batchResultId=aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa[39m[22m

[1mContinuous Testing Summary:[22m
Test Results: [32m[1m3[22m passed[39m, [31m[1m0[22m failed[39m, [33m[1m1[22m failed (non-blocking)[39m ([31m[1m1[22m critical errors[39m)
Total Duration: 9m 28s"
`;

exports[`Default reporter runEnd Complex case with all the tests and results outcomes possible 1`] = `
"[33m[1m2[22m tests not found[39m [90m(ccc-ccc-ccc, ddd-ddd-ddd)[39m
View full summary in Datadog: [2m[36mhttps://app.datadoghq.com/synthetics/explorer/ci?batchResultId=batch-id[39m[22m

[1mContinuous Testing Summary:[22m
Test Results: [32m[1m2[22m passed[39m, [31m[1m1[22m failed[39m, [33m[1m3[22m failed (non-blocking)[39m, [1m1[22m skipped ([33m[1m1[22m timed out[39m, [31m[1m2[22m critical errors[39m)
Total Duration: 9m 28s"
`;

exports[`Default reporter runEnd Simple case with 1 test with 1 result (passed) 1`] = `
"View full summary in Datadog: [2m[36mhttps://app.datadoghq.com/synthetics/explorer/ci?batchResultId=aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa[39m[22m

[1mContinuous Testing Summary:[22m
Test Results: [32m[1m1[22m passed[39m, [31m[1m0[22m failed[39m
Total Duration: 9m 28s"
`;

exports[`Default reporter runEnd communicates 2 tests parallelization 1`] = `
"View full summary in Datadog: [2m[36mhttps://app.datadoghq.com/synthetics/explorer/ci?batchResultId=aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa[39m[22m

[1mContinuous Testing Summary:[22m
Test Results: [32m[1m1[22m passed[39m, [31m[1m0[22m failed[39m
Max parallelization configured: 2 tests running at the same time
Total Duration: 9m 28s

Increase your parallelization to reduce your total duration: [2m[36mhttps://app.datadoghq.com/synthetics/settings/continuous-testing[39m[22m
"
`;

exports[`Default reporter runEnd communicates no parallelization (1 test at a time) 1`] = `
"View full summary in Datadog: [2m[36mhttps://app.datadoghq.com/synthetics/explorer/ci?batchResultId=aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa[39m[22m

[1mContinuous Testing Summary:[22m
Test Results: [32m[1m1[22m passed[39m, [31m[1m0[22m failed[39m
Max parallelization configured: 1 test running at the same time
Total Duration: 9m 28s

Increase your parallelization to reduce your total duration: [2m[36mhttps://app.datadoghq.com/synthetics/settings/continuous-testing[39m[22m
"
`;

exports[`Default reporter runEnd does not communicate for uncapped orgs 1`] = `
"View full summary in Datadog: [2m[36mhttps://app.datadoghq.com/synthetics/explorer/ci?batchResultId=aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa[39m[22m

[1mContinuous Testing Summary:[22m
Test Results: [32m[1m1[22m passed[39m, [31m[1m0[22m failed[39m
Total Duration: 9m 28s"
`;

exports[`Default reporter testTrigger Blocking test, with 1 config override 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Found test "[32m[1mRequest on example.org[22m[39m" [90m(1 config override)[39m
"
`;

exports[`Default reporter testTrigger Blocking test, with 2 config overrides 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Found test "[32m[1mRequest on example.org[22m[39m" [90m(2 config overrides)[39m
"
`;

exports[`Default reporter testTrigger Blocking test, without config overwrite 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Found test "[32m[1mRequest on example.org[22m[39m"
"
`;

exports[`Default reporter testTrigger Non-blocking test from Datadog, with 1 config override 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Found test "[32m[1mRequest on example.org[22m[39m" (non-blocking) [90m(1 config override)[39m
"
`;

exports[`Default reporter testTrigger Non-blocking test from Datadog, with 2 config overrides 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Found test "[32m[1mRequest on example.org[22m[39m" (non-blocking) [90m(2 config overrides)[39m
"
`;

exports[`Default reporter testTrigger Non-blocking test from Datadog, without config overwrite 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Found test "[32m[1mRequest on example.org[22m[39m" (non-blocking)
"
`;

exports[`Default reporter testTrigger Non-blocking test, with 1 config override 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Found test "[32m[1mRequest on example.org[22m[39m" (non-blocking) [90m(1 config override)[39m
"
`;

exports[`Default reporter testTrigger Non-blocking test, with 2 config overrides 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Found test "[32m[1mRequest on example.org[22m[39m" (non-blocking) [90m(2 config overrides)[39m
"
`;

exports[`Default reporter testTrigger Skipped test from Datadog, with 1 config override 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Skipped test "[33m[2mRequest on example.org[22m[39m" because of execution rule configuration in Datadog
"
`;

exports[`Default reporter testTrigger Skipped test from Datadog, with 2 config overrides 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Skipped test "[33m[2mRequest on example.org[22m[39m" because of execution rule configuration in Datadog
"
`;

exports[`Default reporter testTrigger Skipped test from Datadog, without config overwrite 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Skipped test "[33m[2mRequest on example.org[22m[39m" because of execution rule configuration in Datadog
"
`;

exports[`Default reporter testTrigger Skipped test, with 1 config override 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Skipped test "[33m[2mRequest on example.org[22m[39m"
"
`;

exports[`Default reporter testTrigger Skipped test, with 2 config overrides 1`] = `
"[[1m[2maaa-bbb-ccc[22m[22m] Skipped test "[33m[2mRequest on example.org[22m[39m"
"
`;

exports[`Default reporter testsWait outputs triggered tests 1`] = `
"View pending summary in Datadog: [2m[36mhttps://app.datadoghq.com/synthetics/explorer/ci?batchResultId=123[39m[22m


- Waiting for [1m[36m11[39m[22m test results [90m(abc-def-ghi, abc-def-ghi, abc-def-ghi, abc-def-ghi, abc-def-ghi, abc-def-ghi, abc-def-ghi, abc-def-ghi, abc-def-ghi, abc-def-ghi, â€¦)[39mâ€¦

"
`;
